### PR TITLE
Fixes #2514 - Updates tests.md for outdated links

### LINKED ISSUE #2514: Update tests.md for changes to browser testing and outdated links
To be fixed:

- "See dev env setup" link under [Functional Tests](https://github.com/webcompat/webcompat.com/blob/31db9c976d5824abd079568a1bce216b4656a598/docs/tests.md#functional-tests) is dead
- Text after `npm test` command  contradicts new headless browser default: 

> ...you should see the browser open and various pages appear and disappear automatically for a minute or two. The tests are complete when the browser window closes...

- Testing Flask link under [Python Unit Tests](https://github.com/webcompat/webcompat.com/blob/31db9c976d5824abd079568a1bce216b4656a598/docs/tests.md#python-unit-tests) returns a 404
- The Intern wiki has been killed (Intern claims it's been "moved" to the regular docs, but in my experience their corresponding links are almost always wrong, and much of that sometimes extra information appears to be simply lost 🙄 ) so all the links under [JS Functional Tests](https://github.com/webcompat/webcompat.com/blob/31db9c976d5824abd079568a1bce216b4656a598/docs/tests.md#js-functional-tests) should be updated to point to the updated-yet-not-always-copiously-detailed [docs](https://theintern.io/docs.html#Intern/4/docs/README.md)

### PR DESCRIPTION
**Pull request by Anushi1998**
Fixes #2514 

It updates [tests.md](https://github.com/webcompat/webcompat.com/blob/master/docs/tests.md) to point to correct links.

### CODE REVIEW

**Comment by Anushi1998** [comment]
@laghee @miketaylr

Let me know if any changes are required :)

File: docs/tests.md [hunk 1]
```diff
@@ -151,15 +151,15 @@ Unit tests are preferred for features or functionality that are independent of t
...
 ### JS Functional Tests
 
-Functional tests are written in JavaScript, using [Intern](http://theintern.io/). There's a nice [guide on the Intern wiki](https://github.com/theintern/intern/wiki/Writing-Tests-with-Intern#functional-testing) that should explain enough to get you started.
+Functional tests are written in JavaScript, using [Intern](http://theintern.io/). There's a nice [guide on the Intern wiki](https://theintern.io/docs.html#Intern/4/docs/docs%2Fwriting_tests.md/functional-tests) that should explain enough to get you started.
 
 Important documentation links:
 * [Leadfoot](https://theintern.github.io/leadfoot/): the library that drives the browser (via Selenium).
 * [ChaiJS](http://chaijs.com/api/assert/): the library used for assertions.
-* [Intern wiki](https://github.com/theintern/intern/wiki): contains useful examples.
+* [Intern wiki](https://theintern.io/docs.html#Intern/4/docs/README.md): contains useful examples.
```
Comment (miketaylr) [inline docs/tests.md]: Can we change this to say Intern docs, instead of wiki?

**Comment by Anushi1998** [comment]
@miketaylr I have to make some more changes and also the HEAD of master branch is already ahead of my local repo it was creating another merge commit from pull and when I rebased it and trying pushing it was causing conflicts, I couldn't understand why but rebasing is not including merge commit from pull also. So instead I have done hard reset and made the changes and pushed. Is it ok?

File: docs/tests.md [hunk 2]
```diff
@@ -144,22 +144,22 @@ Contributions that add or modify major functionality to the project should typic
 
 ### Python Unit Tests
 
-Our Python unit tests are vanilla flavored [`unittest`](https://docs.python.org/2/library/unittest.html) tests. Unit tests placed in the `tests` directory will be automatically detected by nose&mdash;no manual registration is necessary.
+Our Python unit tests are vanilla flavored [`unit test`](https://docs.python.org/2/library/unittest.html) tests. Unit tests placed in the `tests` directory will be automatically detected by nose&mdash; no manual registration is necessary.
```
Comment (miketaylr) [inline docs/tests.md]: This one should actually be `unittest` as a single word, since it refers to a Python class.

**Comment by miketaylr** [comment]
> Is it ok?

Yep, looks good -- just a single commit with your changes, and no merge commits.

**Comment by Anushi1998** [comment]
@miketaylr I have updated it. Please have a review :)

**Pull request merged**